### PR TITLE
Init global state in Gatsby v4 compatible way #157

### PR DESCRIPTION
**Pull request by olavea**

### CODE REVIEW

File: packages/gatsby-transformer-cloudinary/gatsby-node.js [hunk 1]
```diff
@@ -10,12 +10,38 @@ const {
...
+} catch (error) {
+  console.error(
+    `Cannot check if Gatsby supports onPluginInit lifecycle: 💜 🐸 on🔌👸 life🚴‍♀️ `
+  );
+}
+
 const pluginOptions = getPluginOptions();
 
-exports.onPreInit = ({ reporter }, pluginOptions) => {
+const initializaGlobalState = (_, pluginOptions, { reporter }) => {
```
Comment (raae) [inline packages/gatsby-transformer-cloudinary/gatsby-node.js]: The params need to be the same as for onPreInit, ie. first deconstruct reporter from the gatsbyUtils, then catch the plugin options.

```
const initializaGlobalState = ({ reporter }, pluginOptions) => {
```

**Review by raae — CHANGES_REQUESTED** [review]
I also see the notice: "The plugin gatsby-transformer-cloudinary@2.2.4 is using the API "unstable_onPluginInit" which is not a known API."

I think this might have to do with us using both gatsby3 and gatsby4 in the same workspace. Will explore further. But you can ignore that warning for now @olavea.

But see the code comment below!

**Comment by raae** [comment]
Also remember to remove the console.logs

File: packages/gatsby-transformer-cloudinary/gatsby-node.js (hunk 1 above)
Comment (olavea) [reply packages/gatsby-transformer-cloudinary/gatsby-node.js:31]: What is the `_,` param?

**Comment by olavea** [comment]
What is the _, param?
Do we need it?

File: packages/gatsby-transformer-cloudinary/gatsby-node.js [hunk 2]
```diff
@@ -10,12 +10,36 @@ const {
...
+  console.error(
+    `Cannot check if Gatsby supports onPluginInit lifecycle: 💜 🐸 on🔌👸 life🚴‍♀️ `
+  );
+}
+
 const pluginOptions = getPluginOptions();
 
-exports.onPreInit = ({ reporter }, pluginOptions) => {
-  setPluginOptions({ pluginOptions, reporter });
+const initializaGlobalState = ({ reporter }, _, pluginOptions) => {
```
Comment (raae) [inline packages/gatsby-transformer-cloudinary/gatsby-node.js]: Remove `, _`

**Review by raae — CHANGES_REQUESTED** [review]

File: packages/gatsby-transformer-cloudinary/gatsby-node.js (hunk 2 above)
Comment (olavea) [reply packages/gatsby-transformer-cloudinary/gatsby-node.js]: 👍

**Pull request closed without merging**